django/db/models/aggregates.py

```diff
@@ -95,7 +95,10 @@ def __init__(
             raise TypeError(f"{self.__class__.__name__} does not allow default.")
 
         self.distinct = distinct
-        self.filter = filter and AggregateFilter(filter)
+        if filter:
+            self.filter = AggregateFilter(filter)
+        else:
+            self.filter = None
```

Comment:
This is from the second commit, fixing the latent issue.

a76035e925ff4e6d8676c65cb135c74b993b1039 changed `BaseExpression.resolve_expression()` to filter source expressions by `if expr is not None` instead of `if expr`. An empty `Q()` is falsey, not `None`. The new code mutating it to `None` would (somehow) lead to incorrect results, detected in admin tests. This commit changes `Aggregate.__init__` to change `filter=Q()` to `filter=None`.

Reply:
I think there's a larger issue lying in the `FILTER (WHERE ...)` support [emulation layer](https://github.com/django/django/blob/4037d8e6a04b0e6f87b8f0626b47ccc555eaafda/django/db/models/aggregates.py#L195-L205) as it doesn't catch `FullResultSet`. I think we should adjust the logic there instead as there are other cases (not only `Q()`) that result in that.

Reply:
Alright, played with the code locally.

The issue stem from a `Q` instance being allowed to be assigned to `self.filter` but calling `filter.__bool__` could have side effects so we should avoid calling it. For example, `filter` could be a queryset returning a single boolean field which would qualify it as a valid conditional expression.

I believe we want the following here to be safe

```suggestion
        self.filter = None if filter is None else AggregateFilter(filter)
```

FTR that's what `OrderByList.from_param` does.